### ISSUE TITLE
Actor vendor_signed_rpm_scanner unexpectedly terminated with exit code: 1

### ISSUE THREAD

**Issue by m-witt** [plesk/centos2alma#171]
**Describe the bug**
Upgrade failed with following error

```
2024-02-08 22:23:31,949 - INFO - ====> * vendor_signed_rpm_scanner
2024-02-08 22:23:31,949 - INFO - Provide data about installed RPM Packages signed by Red Hat.
2024-02-08 22:23:33,520 - INFO - Process Process-320:
2024-02-08 22:23:33,521 - INFO - Traceback (most recent call last):
2024-02-08 22:23:33,521 - INFO - File "/usr/lib64/python2.7/multiprocessing/process.py", line 258, in _bootstrap
2024-02-08 22:23:33,521 - INFO - self.run()
2024-02-08 22:23:33,521 - INFO - File "/usr/lib64/python2.7/multiprocessing/process.py", line 114, in run
2024-02-08 22:23:33,521 - INFO - self._target(*self._args, **self._kwargs)
2024-02-08 22:23:33,521 - INFO - File "/usr/lib/python2.7/site-packages/leapp/repository/actor_definition.py", line 72, in _do_run
2024-02-08 22:23:33,521 - INFO - actor_instance.run(*args, **kwargs)
2024-02-08 22:23:33,521 - INFO - File "/usr/lib/python2.7/site-packages/leapp/actors/__init__.py", line 290, in run
2024-02-08 22:23:33,522 - INFO - self.process(*args)
2024-02-08 22:23:33,522 - INFO - File "/usr/share/leapp-repository/repositories/system_upgrade/common/actors/redhatsignedrpmscanner/actor.py", line 146, in process
2024-02-08 22:23:33,522 - INFO - "Package {} is unsigned, packager: {}, signature: {}".format(pkg.name, pkg.packager, pkg.pgpsig)
2024-02-08 22:23:33,522 - INFO - UnicodeEncodeError: 'ascii' codec can't encode characters in position 9-10: ordinal not in range(128)
2024-02-08 22:23:33,525 - INFO - ==========================================================================================================
2024-02-08 22:23:33,525 - INFO - Actor vendor_signed_rpm_scanner unexpectedly terminated with exit code: 1 - Please check the above details
2024-02-08 22:23:33,525 - INFO - ==========================================================================================================
```

**Feedback archive**
*Please attach a feedback archive to the bug report.*
[centos2alma_feedback.zip](https://github.com/plesk/centos2alma/files/14215129/centos2alma_feedback.zip)

**Comment by SandakovMM** [comment]
Hm, looks like some troubles on the leapp side. Let me see what exactly happens

**Comment by SandakovMM** [comment]
It seems that there is an rpm package installed on the server that is not signed and its name is not in ASCII format. Could you please provide the output of the following command: `rpm -qa --qf '%{NAME}-%{VERSION}-%{RELEASE} %{SIGPGP:pgpsig} %{SIGGPG:pgpsig}\n' | grep "(none).*(none)"`

**Comment by m-witt** [comment]
> It seems that there is an rpm package installed on the server that is not signed and its name is not in ASCII format. Could you please provide the output of the following command: `rpm -qa --qf '%{NAME}-%{VERSION}-%{RELEASE} %{SIGPGP:pgpsig} %{SIGGPG:pgpsig}\n' | grep "(none).*(none)"`

```
gpg-pubkey-8608cb71-4acc7220 (none) (none)
bitstream-vera-fonts-1.10-7 (none) (none)
wkhtmltox-0.12.5-1.centos7 (none) (none)
gpg-pubkey-98661eb8-62e960b0 (none) (none)
xmlrpc-c-c++-1.32.5-1905.svn2451.el7 (none) (none)
gpg-pubkey-c74cd1d8-56fc10ab (none) (none)
gpg-pubkey-914bdf7e-55c05220 (none) (none)
gpg-pubkey-5ebd2744-418ffac9 (none) (none)
sharutils-4.13.3-8.el7 (none) (none)
blackfire-php-1.92.9-1 (none) (none)
gpg-pubkey-2c52609d-55a59f6e (none) (none)
gpg-pubkey-1bb943db-511147a9 (none) (none)
gpg-pubkey-ef8d349f-57b6233e (none) (none)
gpg-pubkey-8507efa5-5b02c2fb (none) (none)
gpg-pubkey-34fa74dd-540237d4 (none) (none)
gpg-pubkey-f4a80eb5-53a7ff4b (none) (none)
gpg-pubkey-4520afa9-50ab914c (none) (none)
gpg-pubkey-703df089-602c07e9 (none) (none)
gpg-pubkey-4bd6ec30-4c37bb40 (none) (none)
gpg-pubkey-352c64e5-52ae6884 (none) (none)
gpg-pubkey-cd2efd2a-4b26dda1 (none) (none)
perl-Text-Iconv-1.7-18.el7 (none) (none)
gpg-pubkey-34410682-5287ce4e (none) (none)
gpg-pubkey-b56a8bac-3bbc4d06 (none) (none)
gpg-pubkey-6963f07f-57fad2ec (none) (none)
gpg-pubkey-9e61ef26-5cabbf8a (none) (none)
inotify-tools-libs-3.14-2 (none) (none)
blackfire-2.25.0-1 (none) (none)
gpg-pubkey-72e0a4f6-58834f84 (none) (none)
gpg-pubkey-621e9f35-58adea78 (none) (none)
gpg-pubkey-24098cb6-58875d52 (none) (none)
```

**Comment by SandakovMM** [comment]
I've identified the package causing the issue, which is **inotify-tools-libs**. The problem is that the packager's representation can't be converted to ASCII. As a result, when Leapp attempted to display it in the logs, it caused a crash:
```
2024-02-08 22:43:38.829 DEBUG    PID: 32395 leapp.workflow.FactsCollection.rpm_scanner: inotify-tools-libs|3.14|2|0|David Hrbáč <david@hrbac.cz>|x86_64|(none)
```
I think removing the package should resolve the problem

We also need to implement a patch for elevate on our end. Additionally, I will reach out to the elevate developers to confirm if the issue is still occurring.

**Comment by SandakovMM** [comment]
The problem should be fixed by #186.
> Additionally, I will reach out to the elevate developers to confirm if the issue is still occurring.

On the elevate side the problem should be already fixed as far as I can see ([there is no such kind of debug anymore](https://github.com/AlmaLinux/leapp-repository/blob/db25c79c414c4f76bb844735f6551237b3f8333c/repos/system_upgrade/common/actors/redhatsignedrpmscanner/actor.py#L78-L92))
